lib/src/parsers/oa3server-parser.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ClassDeclaration,
+  MethodDeclaration,
+  ParameterDeclaration
+} from "ts-morph";
+import { Oa3Server, Oa3ServerVariable } from "../definitions";
+import {
+  getDecoratorConfigOrThrow,
+  getJsDoc,
+  getObjLiteralPropOrThrow,
+  getParameterPropertySignaturesOrThrow,
+  getParamWithDecorator,
+  getPropertyName,
+  getPropValueAsStringOrThrow
+} from "./parser-helpers";
+import { Oa3serverConfig } from "../syntax/oa3server";
+import { parseType } from "./type-parser";
+import { TypeTable } from "../types";
+import { LociTable } from "../locations";
+import { err, ok, Result } from "../util";
+import { OptionalNotAllowedError, ParserError } from "../errors";
+
+export function parseOa3Servers(
+  klass: ClassDeclaration,
+  typeTable: TypeTable,
+  lociTable: LociTable
+): Result<Oa3Server[], ParserError> {
+  const serverMethods = klass
+    .getMethods()
+    .filter(m => m.getDecorator("oa3server") !== undefined);
+
+  const servers: Oa3Server[] = [];
+  for (const method of serverMethods) {
+    const serverResult = parseOa3Server(method, typeTable, lociTable);
+    if (serverResult.isErr()) return serverResult;
+    servers.push(serverResult.unwrap());
+  }
+  return ok(servers);
+}
+
+export function parseOa3Server(
```

Comment:
Is this function is anywhere else? If not, we can remove the `export` declaration